xapian-core/weight/tfidfweight.cc

```diff
@@ -70,26 +74,48 @@ TfIdfWeight::name() const
 string
 TfIdfWeight::serialise() const
 {
-    return normalizations;
+    if (param_s > 0 || param_delta > 0) {
+	string result = serialise_double(param_s);
+	result += serialise_double(param_delta);
+	return result;
+    } else {
+	return normalizations;
+    }
 }
 
 TfIdfWeight *
-TfIdfWeight::unserialise(const string & s) const
+TfIdfWeight::unserialise(const string & str) const
```

Comment:
Was there a need to change the variable name? It makes it slightly harder to see what changes have been made (here it's not hugely important, but in a more complex function or change it might be).